### PR TITLE
[code-input] Add support for GROQ

### PR DESCRIPTION
**Pull request by kristofferj**
**Type of change (check at least one)**
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Documentation (fix or update to documentation)
- [ ]  Maintenance
- [ ]  Other, please describe:

**Does this change require a documentation update? (Check one)**
- [ ]  Yes
- [x]  No

**Current behavior**
If a user adds a custom language, it is filtered out.
Changes this to fallback with text

**Note for release**
- GROQ support in code-input. Also let you define custom language (without syntax highlighting)

### CODE REVIEW

File: packages/@sanity/code-input/src/CodeInput.js [hunk 1]
```diff
@@ -212,10 +213,9 @@ export default class CodeInput extends PureComponent {
       if (!SUPPORTED_LANGUAGES.find(lang => lang.value === value)) {
         // eslint-disable-next-line no-console
         console.warn(
-          `'options.languageAlternatives' lists a language which is not supported: "%s", ignoring it.`,
+          `'options.languageAlternatives' lists a language which is not supported: "%s", Syntax highlighting will be disabled.`,
```
Comment (thomax) [inline packages/@sanity/code-input/src/CodeInput.js]: Syntax --> syntax

File: packages/@sanity/code-input/src/CodeInput.js [hunk 2]
```diff
@@ -265,9 +265,8 @@ export default class CodeInput extends PureComponent {
       )
     }
 
-    const isSupported = isSupportedLanguage(value && value.language)
     const selectedLanguage =
-      value && value.language ? languages.find(item => item.value === isSupported) : undefined
+      value && value.language ? languages.find(item => item.value === value.language) : undefined
```
Comment (thomax) [inline packages/@sanity/code-input/src/CodeInput.js]: I'm not sure why isSupported isn't called here anymore?

File: packages/@sanity/code-input/src/groq.js [hunk 3]
```diff
@@ -0,0 +1,611 @@
+/* eslint-disable no-undef */
+const rules = {
```
Comment (rexxars) [inline packages/@sanity/code-input/src/groq.js]: Maybe add a comment to signal where this definition is coming from, to make it easier to keep in sync?

```suggestion
// Grammar from https://github.com/sanity-io/vscode-sanity
const rules = {
```

File: packages/@sanity/code-input/src/groq.js [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+}
+
+ace.define(
+  'ace/mode/groq_highlight_rules',
+  ['require', 'exports', 'module', 'ace/lib/oop', 'ace/mode/text_highlight_rules'],
+  function(acequire, exports, module) {
+    const oop = acequire('../lib/oop')
+    const TextHighlightRules = acequire('./text_highlight_rules').TextHighlightRules
+
+    const QroqHighlightRules = function() {
```
Comment (rexxars) [inline packages/@sanity/code-input/src/groq.js]: `Qroq` or `Groq`? :)

Comment (kristofferj) [reply packages/@sanity/code-input/src/groq.js]: oups :)

**Pull request merged**